builder/amazon/common/step_security_group.go

```diff
@@ -55,8 +55,12 @@ func (s *StepSecurityGroup) Run(state multistep.StateBag) multistep.StepAction {
 	group := &ec2.CreateSecurityGroupInput{
 		GroupName:   &groupName,
 		Description: aws.String("Temporary group for Packer"),
-		VpcId:       &s.VpcId,
 	}
+
+	if s.VpcId != "" {
```

Comment:
vpc id should never be an empty string after https://github.com/hashicorp/packer/pull/4879